### ISSUE TITLE
[web] Keyboard navigation between text fields using TAB key does not work

### ISSUE THREAD

**Issue by aryzhov** [flutter/flutter#32778]
To reproduce, open Flutter Web Gallery demo, for example [here](https://flutter-web-demo.firebaseapp.com/#/material/text-form-field). Go to Material > Text Fields page. Place the cursor on the first field. Press the TAB key. Expected behavior: the cursor moves to the next form field. Actual behavior: the cursor moves to the browser menu. 

## flutter doctor
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, v1.5.8, on Mac OS X 10.13.6 17G6030, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[!] iOS toolchain - develop for iOS devices (Xcode 10.1)
    ✗ ios-deploy out of date (1.9.4 is required). To upgrade with Brew:
        brew upgrade ios-deploy
[✓] Android Studio (version 3.4)
[✓] VS Code (version 1.33.1)
[!] Connected device
    ! No devices available

! Doctor found issues in 2 categories.
```

**Comment by rvowles** [comment]
It would be nice if it was known if there was an intention to fix this or we should put the effort into working around it.

**Comment by EricPreventa** [comment]
Likewise here. Is there something that needs to be done to fix this?

**Comment by rishisinghal** [comment]
Yes this is an important functionality for any forms as users are accustomed to using tab. 
The **Hack** I tried today which worked was using 
                 RawKeyboardListener(
                    focusNode: _emailFocus,
                    onKey: (key) {
                      RawKeyEventDataAndroid kData = key.data;
//                      print('Key code: ${kData.keyCode}');
                      if(kData.keyCode==9) {
                        FocusScope.of(context).requestFocus(_passwordFocus);
                      }
                    },

This was causing it to go to the next field but also to the browser url field. Then I tried the below (to prevent the default behaviour since I did not find any preventDefault function) in my index.html and it worked
        document.addEventListener('keydown', function(event) {
            // console.log('event key'+event.code);
            if (event.code == 'Tab') {
                event.preventDefault();
            }
        });

**Comment by EricPreventa** [comment]
thanks @rishisinghal for sharing this. I'll be trying to implement this shortly. This is definitely a "hack" as I feel long forms it'll certainly affect the code readability, but it doesn't look like we have much of an option otherwise.

**Comment by zanecidan** [comment]
> Yes this is an important functionality for any forms as users are accustomed to using tab.
> The **Hack** I tried today which worked was using
> RawKeyboardListener(
> focusNode: _emailFocus,
> onKey: (key) {
> RawKeyEventDataAndroid kData = key.data;
> // print('Key code: ${kData.keyCode}');
> if(kData.keyCode==9) {
> FocusScope.of(context).requestFocus(_passwordFocus);
> }
> },
> 
> This was causing it to go to the next field but also to the browser url field. Then I tried the below (to prevent the default behaviour since I did not find any preventDefault function) in my index.html and it worked
> document.addEventListener('keydown', function(event) {
> // console.log('event key'+event.code);
> if (event.code == 'Tab') {
> event.preventDefault();
> }
> });

I tried this workaround but was getting 'Tried to make a child into a parent itself' error.

**Comment by stuartmorgan** [comment]
This is a duplicate of #1608, and there's active work going on in that area.
